doc/library/scan.txt

```diff
@@ -361,6 +361,46 @@ If we do not pass the ``updates`` dictionary to the function, then
 ``a.value`` will always remain 1, ``b`` will always be 2 and ``c``
 will always be ``12``.
 
+
+Using shared variables - the strict flag
+----------------------------------------
+
+You also have the possibility to use the ``strict`` flag. When set to true,
+Scan assumes that all the necessary shared variables in ``fn`` are passed as a
+part of ``non_sequences``. This has to be ensured by the user. Otherwise, it
+will result in an error. This avoids Scan Op calling any earlier (external) Op
+over and over, and speeds up the process.
```

Comment:
Can you specify if this cause scan grad speed up, scan optimization speed up or scan execution speed up? All of them?